### ISSUE TITLE
Support for upgradeStrategy

### ISSUE THREAD

**Issue by nm-harry** [nutmegdevelopment/marathon-config-generator#1]
Doesn't seem to support generation of upgradeStrategy from YML file.

```
"upgradeStrategy": { }
```

```
upgradeStrategy:
  minimumHealthCapacity: 0
  maximumOverCapacity: 0
```

**Comment by nm-harry** [comment]
Looking at this, the `omitempty` on the yaml > json will ignore anything that is zero, but a pointer can have a present zero value. I've tried the following locally and it adds `upgradeStrategy` when there are 0 values.

```go
UpgradeStrategy       struct {
		MaximumOverCapacity   *float64 `yaml:"maximumOverCapacity" json:"maximumOverCapacity,omitempty"`
		MinimumHealthCapacity *float64 `yaml:"minimumHealthCapacity" json:"minimumHealthCapacity,omitempty"`
	} `yaml:"upgradeStrategy" json:"upgradeStrategy,omitempty"`
```

```json
"upgradeStrategy": {
  "maximumOverCapacity": 0.25,
  "minimumHealthCapacity": 0.3
}
```

**Comment by nm-harry** [comment]
Merged branch upgrade-fix efb1a5c201f92f4433a35bc1297435a811ec8359 that implements the above change. Generated config can now use 0 values for upgradeStrategy and produce JSON that is not empty.
